### DISCUSSION TITLE
Generate fake data from Entity

### DISCUSSION THREAD

**Discussion by dephiros** [coinbase/rest-hooks#2594] [Q&A]
Hi,

Found this lib and really impressed with all the thinking and works here.

At my company, we try to find a way generate faker obj with `Entity`. Is there a  recommended way to specify some meta data like `faker` field with `Entity` property? Maybe decorator?

Thanks

**Comment by ntucker** [comment]
Not exactly sure what you mean, but you can mock data with [fixtures and interceptors](https://resthooks.io/docs/api/Fixtures). If you want to create a random Entity instance, you can use [Entity.fromJS](https://resthooks.io/rest/api/Entity#fromJS) like:

```ts
const instance = MyEntity.fromJS({title: 'my title'});
```

**Comment by dephiros** [comment]
Hi,

Thank you for your response. I guess a better phrasing of the question would be a good way to associate extra meta data with each field. Right now, I see that we have `static schema` to define an object schema. Maybe to use with faker, we would

```
class Todo extends Entity {
  readonly id: number = 0;
  readonly user: User = User.fromJS({});
  @fake()
  readonly title: string = '';
  readonly completed: boolean = false;

  pk() {
    return `${this.id}`;
  }

  static faker = {
    id: "datatype.uuid"
    title: "lorem.lines"
  }

  static schema = {
    user: User,
  };
}
```
And maybe define a function `createFakeEntity(Todo)` that would generate these entity using the `faker` metadata. Is a `static` field what you recommend as well for adding extra metadata?
I have seen other lib like django using a class to represent Field or some others use `decorator`

**Comment by dephiros** [comment]
Ok, revised my strategy. Instead of generate faker from `Entity`, I decided to generate faker from JSON schema
- Use JSON Schema tooling
- Can use `MyEntity.fromJS`.
Thank you for your help

**Reply by ntucker** [reply]
That makes sense - I'm glad you figured out a path forward.

This library is focused on integrating the data; so as long as we can make hooking into other tools easy I think it's best to leave those things to the related libraries and just make sure they work super well together.

If you see any common patterns repeating about this integration, I'm happy to integrate designs that make this easier as well!
